samples/sample_fei/src/sample_fei.cpp

```diff
@@ -1244,13 +1243,18 @@ mfxStatus ParseParFile(FILE *parFile, std::vector<AppConfig*> &config, msdk_char
         }
         if(!argc)
             continue;
-        for(int i = 0; i < argc; i++)
+
+        for(mfxU32 i = 0; i < argc; i++)
         {
             msdk_printf(MSDK_STRING("%s "), argv[i]);
         }
-        config.push_back(new AppConfig);
-        sts = ParseInputString(argv, (mfxU8)argc, config[configCount]);
+
+        std::unique_ptr<AppConfig> tmp(new AppConfig);
+        config.emplace_back(std::move(tmp));
```

Comment:
You should just pass new AppConfig to emplace_back to avoid local.

Reply:
No, it's not exception safe way, see here http://kayari.org/cxx/antipatterns.html#emplace-new
If it so happens that container is full and reallocation is required, dynamic memory could be lost. I haven't found strict requirement for compiler to call new element constructor first before container resizing, so consider this is the safest way. Also it is possible to get rid of tmp and use rvalue directly at emplace_back but that would overcomplicate the code line. Maybe make_unique is the best option here, but I'm not sure that we support c++14 features